tests/Backend/Mixed/Workspaces/S3WorkspacesTest.php

```diff
@@ -615,59 +349,200 @@ public function testLoadIncrementalNotNullable($backend, $bucketBackend)
         }
     }
 
-    public function workspaceBackendData()
+    public function testDataTypesLoadToRedshift()
     {
-        return [
-            [self::BACKEND_SNOWFLAKE, ["amount" => "NUMBER"]],
-            [self::BACKEND_REDSHIFT, ["amount" => "INT"]],
+
+        $bucketBackend = self::BACKEND_SNOWFLAKE;
+
+        if ($this->_client->bucketExists("out.c-mixed-test-" . $bucketBackend)) {
+            $this->_client->dropBucket(
+                "out.c-mixed-test-{$bucketBackend}",
+                [
+                    'force' => true,
+                ]
+            );
+        }
+
+        if ($this->_client->bucketExists("in.c-mixed-test-" . $bucketBackend)) {
+            $this->_client->dropBucket("in.c-mixed-test-{$bucketBackend}", [
+                'force' => true,
+            ]);
+        }
+        $bucketId = $this->_client->createBucket("mixed-test-{$bucketBackend}", "in", "", $bucketBackend);
+
+        //setup test table
+        $this->_client->createTable(
+            $bucketId,
+            'dates',
+            new CsvFile(__DIR__ . '/../../../_data/dates.csv')
+        );
+
+        $workspaces = new Workspaces($this->_client);
+
+        $workspace = $workspaces->createWorkspace([
+            'backend' => self::BACKEND_REDSHIFT,
+        ]);
+
+        $options = [
+            "input" => [
+                [
+                    "source" => "in.c-mixed-test-{$bucketBackend}.dates",
+                    "destination" => "dates",
+                    "columns" => [
+                        [
+                            'source' => 'valid_from',
+                            'type' => 'TIMESTAMP',
+                        ],
+                    ],
+                ],
+            ],
         ];
+
+        // exception should not be thrown, date conversion should be applied
+        $workspaces->loadWorkspaceData($workspace['id'], $options);
+
+        $wsBackend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $data = $wsBackend->fetchAll("dates", \PDO::FETCH_ASSOC);
+        $this->assertCount(3, $data);
```

Comment:
Možná by stálo za to zkontrolovat, že se ty data zkonvertovali správně?

Reply:
Původní testy já sem to enom refaktoroval.

Reply:
OK